src/libstd/ffi/c_str.rs

```diff
@@ -522,6 +522,37 @@ impl ToOwned for CStr {
     }
 }
 
+#[unstable(feature = "cstring_asref", reason = "recently added", issue = "0")]
+impl ops::Index<ops::RangeFull> for CString {
```

Comment:
`Index<RangeTo>` and `Index<Range>` can't really work, you would have to insert a 0 byte in the middle of the string. And implementing only `Index<RangeFrom>` also seems strange (though potentially useful).